service/src/main/kotlin/xyz/lebkuchenfm/api/auth/ValidateAuthHandler.kt

```diff
@@ -18,74 +15,73 @@ import xyz.lebkuchenfm.domain.users.SetPasswordError
 
 class ValidateAuthHandler(private val authService: AuthService) {
     suspend fun credentialsHandler(credentials: UserPasswordCredential, call: ApplicationCall): UserSession? {
-        return authService.authenticateWithCredentials(credentials.name, credentials.password)
+        return authService
+            .authenticateWithCredentials(credentials.name, credentials.password)
             .onFailure { error ->
-                val status = HttpStatusCode.Unauthorized
-                val instance = call.request.uri
-                val response: ProblemResponse = when (error) {
-                    AuthError.BadCredentialsError -> ProblemResponse(
-                        "Could not authenticate.",
-                        "Wrong password was provided.",
-                        status,
-                        instance,
-                    )
+                when (error) {
+                    AuthError.BadCredentialsError -> {
+                        call.respondWithProblem(
+                            title = "Could not authenticate.",
+                            detail = "Wrong password was provided.",
```

Comment:
`Wrong password was provided.`
or 
`Wrong username or password was provided.`

we should not indicate that username exists when user typed wrong credentials